enhancements/user-defined-default-ingress-controller.md

```diff
@@ -0,0 +1,262 @@
+during OpenShift installation.
+
+Changes to the OpenShift [Installer](https://github.com/openshift/installer) and
+[Ingress Operator](https://github.com/openshift/cluster-ingress-operator) are
+proposed which will provide the basis for a supportable customization procedure.
+The procedure is proposed for incorporation into the OpenShift
+[documentation](https://github.com/openshift/openshift-docs).
+
+## Motivation
+
+OpenShift's default IngressController is automatically created based on the
+cluster platform and cluster-global configuration. The default is reliable for
+installation, but typically requires reconfiguration soon after installation.
+
+However, there are some updates to an existing IngressController which are
+disruptive or destructive and which users would prefer to configure during
+installation.
+
+Because of the broad range of ingress needs, it's difficult to anticipate what
+subset of IngressController API fields should be exposed for configuration at
+the time of installation  Allowing the user to replace the default
+IngressController in its entirety provides maximum expressiveness and
+flexibility without introducing new configuration API surface area.
+
+### Goals
+
+* Create a procedure which safely allows users to provide a specific default
+  IngressController resource to the installer.
+
+### Non-Goals
+
+* No changes to the `config.openshift.io` or `operator.openshift.io` APIs are
+  proposed.
+
+## Proposal
+
+The following specific changes are proposed.
+
+1. Add a `render` command to the Ingress Operator binary which emits the
+   following manifests:
+    * The IngressController [Custom Resource Definition](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/00-custom-resource-definition.yaml).
+    * The `openshift-ingress-operator`
+      [Namespace](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/00-namespace.yaml).
+2. Update the [Installer manifest rendering](https://github.com/openshift/installer/blob/master/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template)
+   system to execute the Ingress Operator `render` command and move the output
+   into the correct location, causing the installer to apply the manifests
+   before the Ingress Operator's lifecycle begins.
+3. Add a installation procedure to the
+   [documentation](https://github.com/openshift/openshift-docs) describing how
+   to provide a default IngressController and reasons why one might want to do
+   so.
+
+### User Stories
+
+Generally speaking, the proposed capability generically improves support for
+many post-installation IngressController use cases. This section describes two
+possible use cases.
+
+#### Create an internal (non-public) cluster
```

Comment:
there is a another enhancement for this https://github.com/openshift/enhancements/pull/25 , _i'm biased, but is more generic and not limited to ingress_

Reply:
Addressed directly in https://github.com/openshift/enhancements/pull/31/files#diff-40bb87d9f5690ccc8c6d75986d956f45R234.

I agree that https://github.com/openshift/enhancements/pull/25 is effective for this use case.

Would it be possible to backport that new API?

Reply:
> there is a another enhancement for this #25 , _i'm biased, but is more generic and not limited to ingress_

We're still looking for evidence of this.  Based on our phone call, the need appears to be restricted to the default ingress operator and nothing else.